File: apps/api/src/oneepis_api/api/v1/routes/ai.py
from typing import Annotated

from fastapi import APIRouter, Depends

from oneepis_api.api.deps import AiAccessDep
from oneepis_api.core.config import Settings, get_settings
from oneepis_api.schemas.ai import AIProviderStatus, ClinicalInsightRequest, ClinicalInsightResponse
from oneepis_api.services.ai.provider import get_ai_provider

router = APIRouter(prefix="/ai", tags=["ai"])
SettingsDep = Annotated[Settings, Depends(get_settings)]


@router.post("/clinical-insights", response_model=ClinicalInsightResponse)
def create_clinical_insight(
    payload: ClinicalInsightRequest,
    settings: SettingsDep,
    _user: AiAccessDep,
) -> ClinicalInsightResponse:
    provider = get_ai_provider(settings)
    return provider.create_insight(payload)


@router.get("/status", response_model=AIProviderStatus)
def get_ai_status(settings: SettingsDep, _user: AiAccessDep) -> AIProviderStatus:
    provider = get_ai_provider(settings)
    status = provider.status()
    if settings.environment.strip().lower() != "development":
        return status.model_copy(update={"base_url": None})
    return status

File: apps/web/src/lib/api/ai.ts
import { apiFetch } from "@/lib/api/client";
import type {
  AIProviderStatus,
  ClinicalInsightRequest,
  ClinicalInsightResponse,
} from "@/lib/types";

export function getAiStatus() {
  return apiFetch<AIProviderStatus>("/api/v1/ai/status");
}

export function createClinicalInsight(payload: ClinicalInsightRequest) {
  return apiFetch<ClinicalInsightResponse>("/api/v1/ai/clinical-insights", {
    method: "POST",
    body: JSON.stringify(payload),
  });
}
